odoo/models.py

```diff
@@ -2756,74 +2756,96 @@ def _inherits_join_calc(self, alias, fname, query):
         :param query: query object on which the JOIN should be added
         :return: qualified name of field, to be used in SELECT clause
         """
+        sql = self._field_to_sql(alias, fname, query)
+        return self.env.cr.mogrify(sql).decode()
+
+    def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None) -> SQL:
```

Comment:
I did not change the `Union` syntax because of _forward references_.  `"SQL" | str` does not work, and the alternative is a bit cumbersome (and actually less readable IMO):
```py
SQLType = typevar("SQLType", bound="SQL")

class SQL:
    def __new__(cls, code: str | SQLType = "", *args, **kwargs):
```
And as for `Optional[X]` vs `X | None`, I don't know which one is better.  The Python doc recommends  `X | Y` instead of `Union[X, Y]`, but there's no explicit recommendation for `Optional`...

Reply:
I wasn't aware of that one, thanks!